### PR TITLE
Add yapf to pre-commit and CI

### PR DESCRIPTION
**Pull request by andrewthetechie**
Resolves https://github.com/TheItsProjects/ItsPrompt/issues/21

This adds a pre-commit hook for yapf to run it on every commit. To use this, make sure pre-commit is installed then run `pre-commit install`. I did not see a good place to setup dev dependencies separate from the main requirements, so I did not add pre-commit anywhere.

It also adds a github action that runs yapf on every push.

### CODE REVIEW

**Comment by TheItsNameless** [comment]
This looks great, do you have any idea how there would be a good way to add dev requirements?

**Comment by andrewthetechie** [comment]
There are a couple of options. The most straight forward would be a file named something like 'requirements-dev.txt' that contain packages only used during development.

If you move to use [poetry for dependency management](https://python-poetry.org/), you could put your dev dependencies in a separate group (`poetry add --group dev pacakge`)

**Comment by TheItsNameless** [comment]
Poetry might be worth a thought, for now I would just add it to a `requirements-dev.txt`.

File: .github/workflows/yapf.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+jobs:
+  formatting-check:
+    name: Formatting Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: run YAPF to test if python code is correctly formatted
+      uses: AlexanderMelde/yapf-action@v1.0
+      with:
+        args: --verbose
```
Comment (TheItsNameless) [inline .github/workflows/yapf.yml:12]: I am not sure about this action, does it run yapf in the main directory? If so, wouldn't it be better to specify the `ItsPrompt` directory by adding `./ItsPrompt/ --recursive` to the args?

**Review by TheItsNameless — CHANGES_REQUESTED** [review]

File: .github/workflows/yapf.yml (hunk 1 above)
Comment (andrewthetechie) [reply .github/workflows/yapf.yml:12]: Here [are the docs](https://github.com/AlexanderMelde/yapf-action) 

The action's [entrypoint is using](https://github.com/AlexanderMelde/yapf-action/blob/master/entrypoint.sh#LL4C1-L4C53
) `--recursive $GITHUB_WORKSPACE`. $GITHUB_WORKSPACE [resolves to the default working directory ](https://docs.github.com/en/actions/learn-github-actions/variables) on the runner for steps, and the default location of your repository when using the [checkout](https://github.com/actions/checkout) action

Comment (TheItsNameless) [reply .github/workflows/yapf.yml:12]: So it looks like there is no action needed, am I right?

Comment (TheItsNameless) [reply .github/workflows/yapf.yml:12]: Well, I think that is done. Will now resolve this and merge it.

**Review by TheItsNameless — APPROVED** [review]

**Pull request merged**